edi_oca/models/edi_configuration.py

```diff
@@ -0,0 +1,193 @@
+            dt = to_utc(dt)
+        return fields.Datetime.to_string(dt)
+
+    def _time_utils(self):
+        return {
+            "datetime": safe_eval.datetime,
+            "dateutil": safe_eval.dateutil,
+            "time": safe_eval.time,
+            "utc_now": fields.Datetime.now(),
+            "date_to_string": self._date_to_string,
+            "datetime_to_string": self._datetime_to_string,
+            "time_to_string": lambda dt: dt.strftime("%H:%M:%S") if dt else "",
+            "first_of": fields.first,
+        }
+
+    def _get_code_snippet_eval_context(self):
+        """Prepare the context used when evaluating python code
+
+        :returns: dict -- evaluation context given to safe_eval
+        """
+        ctx = {
+            "uid": self.env.uid,
+            "user": self.env.user,
+            "DotDict": DotDict,
+            "conf": self,
+        }
+        ctx.update(self._time_utils())
+        return ctx
+
+    def _evaluate_code_snippet(self, snippet, **render_values):
+        if not self._code_snippet_valued(snippet):
+            return {}
+        eval_ctx = dict(render_values, **self._get_code_snippet_eval_context())
+        safe_eval.safe_eval(snippet, eval_ctx, mode="exec", nocopy=True)
+        result = eval_ctx.get("result", {})
+        if not isinstance(result, dict):
+            return {}
+        return result
+
+    def edi_exec_snippet_before_do(self, record, **kwargs):
+        self.ensure_one()
+        # Execute snippet before do
+        vals_before_do = self._evaluate_code_snippet(
+            self.snippet_before_do, record=record, **kwargs
+        )
+
+        # Prepare data
+        vals = {
+            "todo": vals_before_do.get("todo", True),
+            "snippet_do_vars": vals_before_do.get("snippet_do_vars", False),
+            "event_only": vals_before_do.get("event_only", False),
+            "tracked_fields": vals_before_do.get("tracked_fields", False),
+            "edi_action": vals_before_do.get("edi_action", False),
+        }
+        return vals
+
+    def edi_exec_snippet_do(self, record, **kwargs):
+        self.ensure_one()
+
+        old_value = kwargs.get("old_vals", {}).get(record.id, {})
+        new_value = kwargs.get("vals", {}).get(record.id, {})
+        vals = {
+            "todo": True,
+            "record": record,
+            "operation": kwargs.get("operation", False),
+            "edi_action": kwargs.get("edi_action", False),
+            "old_value": old_value,
+            "vals": new_value,
+        }
+        if self.snippet_before_do:
+            before_do_vals = self.edi_exec_snippet_before_do(record, **kwargs)
+            vals.update(before_do_vals)
+        if vals["todo"]:
+            return self._evaluate_code_snippet(self.snippet_do, **vals)
+        return True
+
+    def edi_get_conf(self, trigger, backend=None):
+        domain = [("trigger", "=", trigger)]
+        backend_ids = self.mapped("type_id.backend_id.id")
```

Comment:
what's the point on filtering on all the backends assigned to the current recordset?